.php-cs-fixer.dist.php

```diff
@@ -0,0 +1,34 @@
+<?php
+/**
+ * PHP-CS-Fixer configuration
+ *
+ * (c) Alley <info@alley.co>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @package wp-find-one
```

Comment:
This construction of comment is modeled off of file headers of other PHP libraries, with a few tweaks to match the WordPress core standards for file headers. For example:

- https://github.com/Seldaek/monolog/blob/60ad5183b5e5d6c9d4047e9f3072d36071dcc161/src/Monolog/Logger.php#L1-L10
- https://github.com/sebastianbergmann/phpunit/blob/77354693683de022702c9b95e7fe8d840476f82c/src/Framework/TestSuite.php#L1-L10

So I'm inclined to leave off an SPDX header for consistency with those.

As a personal aside, while the SPDX header might be machine-readable, it doesn't come off to me as super human-readable. If I don't know what SPDX is (I had to Google it), the line won't mean anything to me. Moreover, it seems redundant: Why include both the license name and the invitation to view `LICENSE` in the same block?

Reply:
> without an eye towards the distinctly separate principle of actual documented best practices, consistency reinforces a copy-pasta status quo

Where is it documented that SPDX is a best practice, other than by the SPDX workgroup itself? Got links? 😄

> when the library primarily consists of a single file which can easily be lifted into another project, I would argue that we should expect that the license will never be recirculated in those cases, so the message about some other file becomes useless upon later reference

This convinces me.

> If I don't know what SPDX is (I had to Google it), the line won't mean anything to me.

I'm not exaggerating. Yes, I can make some guesses based on familiar strings, but I wouldn't want to bet the compliance of my code with a license on it. "I won't tell you the license, go find it" is curt but pretty unambiguous in telling you "the answer is not here."

In any case, I'm not quite prepared to make the call by myself to switch to SPDX, so I'll bring this discussion to the open source working group to find an improvement, SPDX or otherwise.

Reply:
Press release: [SPDX Becomes Internationally Recognized Standard for Software Bill of Materials - Linux Foundation](https://linuxfoundation.org/press-release/spdx-becomes-internationally-recognized-standard-for-software-bill-of-materials/)

> _Backed by many of the world’s largest companies for more than a decade, SPDX formally becomes an internationally recognized ISO/IEC JTC 1_ _standard during a transformational time for software and supply chain security_

Also see: [Solving License Compliance at the Source: Adding SPDX License IDs - Linux Foundation](https://www.linuxfoundation.org/blog/solving-license-compliance-at-the-source-adding-spdx-license-ids/)

SPDX identifiers are included in the Wikipedia sidebars for software licenses:

- https://en.wikipedia.org/wiki/GNU_General_Public_License
- https://en.wikipedia.org/wiki/MIT_License
- https://en.wikipedia.org/wiki/Unlicense

Specifically in the context of WordPress:

- https://github.com/WordPress/gutenberg/pull/4338
- https://core.trac.wordpress.org/ticket/43032

In terms of source file headers specifically… I think that's certainly more subjective, since file headers can lend a bit of character to a project. But at the very least I would expect a file header, if there is one, would include the name of the license. Rather than winging it, we have the official license naming schemes provided by SPDX to lean on. The "Solving License Compliance at the Source" article linked above highlights some early examples.